lib/datadog/tracing/span_operation.rb

```diff
@@ -457,8 +445,8 @@ def build_span
           duration: duration,
           end_time: @end_time,
           id: @id,
-          meta: meta.dup,
-          metrics: metrics.dup,
+          meta: Core::Utils::SafeDup.frozen_or_dup(meta),
+          metrics: Core::Utils::SafeDup.frozen_or_dup(metrics),
```

Comment:
Wait... This change isn't equivalent -- the previous version only `dup`'d stuff, whereas now we're also freezing it. I think freezing seems like the right way to go, but worth triple-checking this is the case (which maybe you have, but I wanted to call it out!)

Reply:
@ivoanjo we do not return a frozen object here. `frozen_or_dup` only dups the object if is not frozen already. `frozen_dup` would be the one that would return a frozen duplicate if is the object is not already frozen

Reply:
Urghhh you're right, I read it as being `frozen_dup`. Please ignore my moment of insanity.